src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/Generated/Recipe.cs

```diff
@@ -377,30 +394,64 @@ private void ConfigureBeanstalkEnvironment(Configuration settings, string beanst
 
             if (settings.VPC.UseVPC)
             {
-                optionSettingProperties.Add(new CfnEnvironment.OptionSettingProperty
+                if (settings.VPC.CreateNew)
                 {
-                    Namespace = "aws:ec2:vpc",
-                    OptionName = "VPCId",
-                    Value = settings.VPC.VpcId
-                });
+                    if (AppVpc == null)
+                        throw new InvalidOperationException($"{nameof(AppVpc)} has not been set. The {nameof(ConfigureVpc)} method should be called before {nameof(ConfigureBeanstalkEnvironment)}");
 
-                if (settings.VPC.Subnets.Any())
-                {
+                    optionSettingProperties.Add(new CfnEnvironment.OptionSettingProperty
+                    {
+                        Namespace = "aws:ec2:vpc",
+                        OptionName = "VPCId",
+                        Value = AppVpc.VpcId
+                    });
+
+                    var vpcSubnets = new SortedSet<string>();
```

Comment:
It never makes sense to combine both the private and public subnets. 

If the environment type is loadbalanced then `aws:ec2:vpc,subnets` should be set to the private subnets and `aws:ec2:vpc,ELBSubnets` should be set to the public subnets. If the environment type is single instance then set `aws:ec2:vpc,subnets` to the public subnets.

Right now when I deploy with load balanced the deployment fails because we add both the public and private and that means we have multiple subnets in the same availability zone. ELB does not allow that and fails.

Reply:
Done